pkg/resource/v1/certconfig/error.go

```diff
@@ -0,0 +1,17 @@
+package certconfig
+
+import "github.com/giantswarm/microerror"
+
+// KeyError is an interface that defines contract between Key error
+// check implementations in CRD frameworks and certconfig resource.
+type KeyError interface {
```

Comment:
As comment says, this interface should provide contract for functions that are usually exported in `key/error.go`.

Reply:
FYI we tend to have a `spec.go` for interface and type definitions.